modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/DurableBackgroundCleanupIndexTreeTask.java

```diff
@@ -192,4 +207,118 @@ public DurableBackgroundCleanupIndexTreeTask(
     @Override public String toString() {
         return S.toString(DurableBackgroundCleanupIndexTreeTask.class, this);
     }
+
+    /**
+     * H2Tree that can be used only to destroy it.
+     */
+    public static class H2TreeToDestroy extends H2Tree {
```

Comment:
I didn't understand what this class is for? As I see H2Tree contains the same implementation of methods(temporaryReleaseLock, maxLockHoldTime)

Reply:
This class uses cache group context instead of cache context in these methods' implementation.